src/clp_ffi_js/ir/UnstructuredIrStreamReader.cpp

```diff
@@ -158,6 +158,29 @@ auto UnstructuredIrStreamReader::decode_range(size_t begin_idx, size_t end_idx,
     );
 }
 
+auto UnstructuredIrStreamReader::find_timestamp_last_occurrence(
```

Comment:
Let's deduplicate this with `StructuredIrStreamReader::find_timestamp_last_occurrence`.
You may refer to `generic_decode_range`.